### DISCUSSION TITLE
exp run: rename --temp?

### DISCUSSION THREAD

**Discussion by jorgeorpinel** [treeverse/dvc#5836] [General]
> Extracted from #5814 cc @dmpetrov 
"The `--temp` option name is a bit confusing. I'd prefer something like `--background`."

We actually discussed the term "background" recently (for docs) and decided that it's misleading: `exp run --background` may sound like `$ exp run &` (on Linux). In reality there's no background process, but rather the experiment runs "outside the workspace" — the phrase we use in docs. That said, "temporary" is also not the best concept.

Ideas: `--outside` ,`--foreign`, or `--adj(acent)`.

Or if we're sticking with the current implementation-detail-based name, maybe `--tmp` (no `e`, as in `.dvc/tmp`)

**Comment by jorgeorpinel** [comment]
Users may understand that `--queue` runs in "tmp dirs" though so that's an argument for keeping `--tmp`. E.g. #5800

**Comment by jorgeorpinel** [comment]
Unless `--tmp` runs are made the default again, rel. https://github.com/iterative/dvc/issues/5750#issuecomment-820837713, in which case the option would no longer exist this issue can be closed.

**Comment by dberenbaum** [comment]
Nice synopsis, @jorgeorpinel! It seems like there's too much uncertainty with how these commands will change going forward to make a decision on this right now. Maybe we can just keep the discussion here and revisit later?

**Comment by dberenbaum** [comment]
An alternative solution is to have `--temp` be the default and use `--workspace` or `--debug` when wanting to run in the workspace. See https://github.com/iterative/dvc/issues/5750#issuecomment-820837713.

**Comment by pmrowla** [comment]
Should note here as well that `--temp` used to be the default. It was changed based on feedback from @dmpetrov since running in the workspace seemed much more intuitive than running outside of the workspace.

**Reply by jorgeorpinel** [reply]
> running in the workspace seemed much more intuitive than running outside of the workspace

Based on DVC 1.x I'd agree. But maybe if/when `repro` is phased out and we also introduce the concept of executors (for remote runs) it may no longer be the case? Def for >3.0 though...
